app/models/embedded_ansible_worker/object_management.rb

```diff
@@ -53,4 +54,74 @@ def ensure_host(provider, connection)
       :variables => {'ansible_connection' => "local"}.to_yaml
     ).id
   end
+
+  def ensure_plugin_playbooks_project_seeded(connection)
+    clean_consolidated_plugin_directory
+    copy_plugin_ansible_content
+
+    commit_git_plugin_content
+
+    if project = existing_plugin_playbook_project(connection)
+      update_playbook_project(project)
+    else
+      create_playbook_project(connection)
+    end
+  ensure
+    # we already have 2 copies: one in the gem and one imported into ansible in the project, delete the temporary one
+    clean_consolidated_plugin_directory
+  end
+
+  private
+
+  CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR = Pathname.new("/var/lib/awx_consolidated_source/projects/gem_ansible_content").freeze
+  def clean_consolidated_plugin_directory
+    FileUtils.rm_rf(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+  end
+
+  def copy_plugin_ansible_content
+    FileUtils.mkdir_p(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+
+    #TODO: make this a public api via an attr_reader
+    Vmdb::Plugins.instance.instance_variable_get(:@registered_ansible_content).each do |content|
+      FileUtils.cp_r(Dir.glob("#{content.path}/*"), CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
```

Comment:
I think this is wrong.  I need to know what [namespaced_role_directories and namespaced_playbook_files](https://github.com/ManageIQ/manageiq-design/issues/40) means so we don't have collisions on content.  @syncrou do you have examples of what these values should be for the manageiq-content repo?

Reply:
@syncrou 

1) Note, I'm copying from the `content.path` down.  It's assuming that the `roles_path` is a child directory of the `path`.  I don't know if this is a valid assumption.

```
irb(main):001:0> Vmdb::Plugins.instance.instance_variable_get(:@registered_ansible_content)
=> [#<Vmdb::Plugins::AnsibleContent:0x00000001303a58 @roles_path=#<Pathname:/opt/rh/cfme-gemset/bundler/gems/cfme-content-e5c494e59a7b/content/ansible/roles>, @path=#<Pathname:/opt/rh/cfme-gemset/bundler/gems/cfme-content-e5c494e59a7b/content/ansible>>]
```

I can instead copy them as two separate operations if needed but it was easier to just do `cp_r` so I don't need to copy recursively by myself.


2) With this said, this is how the current `CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR` looks after this method is complete.  It appears to be namespace in that the original filenames are namespaced.  As long as this is the expectation, I think we're ok here.  Right?

```
[root@localhost vmdb]# tree /var/lib/awx_consolidated_source/projects/gem_ansible_content/
/var/lib/awx_consolidated_source/projects/gem_ansible_content/
└── roles
    ├── miq-core.manageiq-automate
    │   ├── action_plugins
    │   │   └── manageiq_automate.py
    │   ├── defaults
    │   │   └── main.yml
    │   ├── examples
    │   ├── handlers
    │   │   └── main.yml
    │   ├── library
    │   │   └── manageiq_automate.py
    │   ├── LICENSE
    │   ├── meta
    │   │   └── main.yml
    │   ├── README.md
    │   ├── tasks
    │   │   └── main.yml
    │   ├── tests
    │   │   ├── inventory
    │   │   └── test.yml
    │   └── vars
    │       └── main.yml
    └── miq-core.manageiq-vmdb
        ├── action_plugins
        │   └── manageiq_vmdb.py
        ├── defaults
        │   └── main.yml
        ├── handlers
        │   └── main.yml
        ├── library
        │   └── manageiq_vmdb.py
        ├── LICENSE
        ├── meta
        │   └── main.yml
        ├── README.md
        ├── tasks
        │   └── main.yml
        ├── tests
        │   ├── inventory
        │   └── test.yml
        └── vars
            └── main.yml
```

Reply:
@jrafanie - That all looks right.  If there were playbooks added to this from other repos we would place them under the `content.path` as well.

Reply:
Ok, sounds good.  Instead of guessing, I'll keep it as is and change when/if we have to.  As long as the gems follow the rules and namespace themselves, this should just work.